src/mappings/mappingUniquesHandlers.ts

```diff
@@ -1,28 +1,145 @@
     uniqueTransfer.txHash = event.extrinsic.extrinsic.hash.toString();
     uniqueTransfer.timestamp = new Date(
-    event.extrinsic.block.timestamp
-  ).getTime();
+      event.extrinsic.block.timestamp
+    ).getTime();
   }
 
+  const collection = await ensureCollection(collectionId.toString());
+  const item = await ensureItem(collectionId.toString(), itemId.toString());
+  item.owner = to.toString();
+  uniqueTransfer.itemId = item.id;
+  uniqueTransfer.collectionId = collection.id;
+
+  await collection.save();
+  await item.save()
   return uniqueTransfer.save();
 }
+
+export const handleUniquesMetadataSetEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug("uniqueMetadataSetEvent added: " + JSON.stringify(event.toHuman()));
+  const collectionId = event.event.data[0];
+  const itemId = event.event.data[1];
+  const data = event.event.data[2];
+  // const isFrozen = event.event.data[3];
+
+  const collection = await ensureCollection(collectionId.toString());
+  const item = await ensureItem(collectionId.toString(), itemId.toString());
+  item.metadataCid = data.toHuman().toString();
+  item.collectionId = collection.id;
+  await collection.save()
+  return item.save();
+}
+
+export const handleUniquesCollectionMetadataSetEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug(
+    "uniqueCollectionMetadataSetEvent added: " + JSON.stringify(event.toHuman())
+  );
+  const collectionId = event.event.data[0];
+  const data = event.event.data[1];
+  // const isFrozen = event.event.data[2];
+
+  const collection = await ensureCollection(collectionId.toString());
+  collection.metadataCid = data.toHuman().toString();
+
+  return collection.save();
+}
+
+export const handleUniquesDestroyedEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug(
+    "handleUniquesDestroyedEvent added: " + JSON.stringify(event.toHuman())
+  );
+  const collectionId = event.event.data[0];
+
+  const collection = await ensureCollection(collectionId.toString());
+  collection.isDestroyed = true;
+  return collection.save()
+}
+
+export const handleUniquesBurnedEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug(
+    "handleUniquesBurnedEvent added: " + JSON.stringify(event.toHuman())
+  );
+  const itemId = event.event.data[1];
+
+  const item = await Item.get(itemId.toString());
+  item.isBurned = true;
+  return item.save();
+}
+
+export const handleUniquesIssuedEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug(
+    "handleUniquesIssuedEvent added: " + JSON.stringify(event.toHuman())
+  );
+  const collectionId = event.event.data[0];
+  const itemId = event.event.data[1];
+  const owner = event.event.data[2];
+
+  const collection = await ensureCollection(collectionId.toString());
+  const item = await ensureItem(collectionId.toString(), itemId.toString());
+  item.owner = owner.toString();
+  item.collectionId = collection.id;
+  item.isBurned = false;
+  await collection.save()
+  return item.save();
```

Comment:
Suggest better highlighting unexpected cases. Something like:
```suggestion
  const collectionId = event.event.data[0].toString();
  const itemId = event.event.data[1].toString();
  const owner = event.event.data[2].toString();

  const storedCollection = await Collection.get(collectionId);
  if (!storedCollection) {
    logger.error(`Handling UniquesIssuedEvent for collection ${collectionId} which doesn't exist!`);
  }

  const storedItem = await Item.get(`${collectionId}-${itemId}`);
  if (storedItem) {
    logger.error(`Handling UniquesIssuedEvent for item ${itemId} which already exists!`);
  }

  const collection = await ensureCollection(collectionId);
  const item = await ensureItem(collectionId, itemId);
  item.owner = owner;

  await collection.save();
  return item.save();
```

Reply:
`ensureCollection` is doing that